redfish-core/lib/cable.hpp

```diff
@@ -0,0 +1,198 @@
+#pragma once
+#include <boost/container/flat_map.hpp>
+#include <utils/json_utils.hpp>
+
+namespace redfish
+{
+using ManagedPropertyType =
+    boost::container::flat_map<std::string, std::variant<std::string, double>>;
+
+/**
+ * @brief Get Cable properties
+ * @param[in,out]   asyncResp   Async HTTP response.
+ * @param[in]       resp        A map to hold Cable properties
+ * @param[in]       cableId     Cable Id
+ */
+inline void
+    getCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       ManagedPropertyType& resp, const std::string& cableId)
+{
+    double* cableLength = nullptr;
+    std::string* cableTypeDescription = nullptr;
+
+    for (auto& propertyMap : resp)
+    {
+        if (propertyMap.first == "CableTypeDescription")
+        {
+            cableTypeDescription =
+                std::get_if<std::string>(&propertyMap.second);
+        }
+        else if (propertyMap.first == "Length")
+        {
+            cableLength = std::get_if<double>(&propertyMap.second);
+        }
+    }
+
+    if (cableTypeDescription == nullptr || cableLength == nullptr)
+    {
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    std::string dataId = "/redfish/v1/Cables/" + cableId;
+    asyncResp->res.jsonValue["@odata.type"] = "#Cable.v1_0_0.Cable";
+    asyncResp->res.jsonValue["@odata.id"] = dataId;
+    asyncResp->res.jsonValue["Id"] = cableId;
+    asyncResp->res.jsonValue["Name"] = "Cable Id";
+    asyncResp->res.jsonValue["LengthMeters"] = *cableLength;
+    asyncResp->res.jsonValue["CableType"] = *cableTypeDescription;
+}
+
+/**
+ * The Cable schema
+ */
+inline void requestRoutesCable(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Cables/<str>/")
+        .privileges({{"Login"}})
```

Comment:
Privileges needs to be filled from "redfish::privileges::"

Reply:
I'm not sure what exactly your ask is here, but assuming to make the changes in privilege_registry.hpp. We are not supposed to be modifying privilege_registry.hpp file directly as it gets overwritten by parse_registries.py script. 
@gtmills any thoughts?

Reply:
Here is one of the privilege entries in chassis.hpp:
.privileges(redfish::privileges::postChassis)

Here is where 'postChassis' is defined (privilege_registry.hpp):
// Chassis
const static auto& getChassis = privilegeSetLogin;
const static auto& headChassis = privilegeSetLogin;
const static auto& patchChassis = privilegeSetConfigureComponents;
const static auto& putChassis = privilegeSetConfigureComponents;
const static auto& deleteChassis = privilegeSetConfigureComponents;
const static auto& postChassis = privilegeSetConfigureComponents;

For Cables if you are expecting to define new entries similar to the above?
If so, it is a 'no no' to add new entries in the privilege_registry.hpp - this is an auto generated file and we are not supposed to modify it. 
Here is the comment in the privilege_registry.hpp file:
#pragma once
// privilege_registry.hpp is generated.  Do not edit directly
...

While it is possible that I'm missing something to follow based on your comment, please be more specific so that I can address it.

Reply:
when you regenerate privilege_registry.hpp with the cables schema included, are there any new lines with any cable specific permissions?

Reply:
@gtmills what is it we have to do for the privileges?

Reply:
There is a new Redfish_1.2.0_PrivilegeRegistry.json 
https://redfish.dmtf.org/registries/

Need to pick up by changing this line and rerunning the script
https://github.com/openbmc/bmcweb/blob/83f984b53eef3277d9d3da292152aec1250f5f4d/scripts/parse_registries.py#L145

It does include cables. 
If you don't mind please push this commit upstream. :)